src/seedu/addressbook/Main.java

```diff
@@ -111,6 +111,10 @@ private CommandResult executeCommand(Command command)  {
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException e) {
+            ui.showToUser(e.getMessage());
+            ui.showToUser("Storage file is read only, changes may not be permanent");
```

Comment:
This is one of the ways of handling the exception.
Think how you can improve the user experience in light of the exception that just occurred, than just showing an error message.

Reply:
I suppose the app can continue running instead of terminating, but only view features will be available?